### PR TITLE
Clean up Bloom index surface and internals

### PR DESCRIPTION
**Pull request by chrisvest**

### CODE REVIEW

File: enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java [hunk 1]
```diff
@@ -37,11 +37,12 @@
     @Test
     public void shouldBeAbleToSpecifyEnglishAnalyzer() throws Exception
     {
+        analyzer = ENGLISH;
         try ( FulltextProvider provider = createProvider() )
         {
-            FulltextFactory fulltextFactory = new FulltextFactory( fs, storeDir, ENGLISH, provider );
-            fulltextFactory.createFulltextIndex( BLOOM_NODES, NODES, singletonList( "prop" ) );
-            provider.init();
+//            FulltextFactory fulltextFactory = new FulltextFactory( fs, storeDir, ENGLISH, provider );
```
Comment (ragadeeshu) [inline enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java]: Leftovers?

File: enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java [hunk 2]
```diff
@@ -67,11 +68,12 @@ public void shouldBeAbleToSpecifyEnglishAnalyzer() throws Exception
     @Test
     public void shouldBeAbleToSpecifySwedishAnalyzer() throws Exception
     {
+        analyzer = SWEDISH;
         try ( FulltextProvider provider = createProvider(); )
         {
-            FulltextFactory fulltextFactory = new FulltextFactory( fs, storeDir, SWEDISH, provider );
-            fulltextFactory.createFulltextIndex( BLOOM_NODES, NODES, singletonList( "prop" ) );
-            provider.init();
+//            FulltextFactory fulltextFactory = new FulltextFactory( fs, storeDir, SWEDISH, provider );
```
Comment (ragadeeshu) [inline enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java]: Leftovers?

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java [hunk 3]
```diff
@@ -57,25 +61,33 @@
...
-     * indexes have been disabled. If so, then the indexes will be rebuilt.
+     * @param fileSystem The filesystem to use.
+     * @param storeDir Store directory of the database.
+     * @param analyzerClassName The Lucene analyzer to use for the {@link LuceneFulltext} created by this factory.
      */
     public FulltextProvider( GraphDatabaseService db, Log log, AvailabilityGuard availabilityGuard,
-                             JobScheduler scheduler, TransactionIdStore transactionIdStore )
+                             JobScheduler scheduler, TransactionIdStore transactionIdStore,
+                             FileSystemAbstraction fileSystem, File storeDir,
+                             String analyzerClassName ) throws IOException
```
Comment (ragadeeshu) [inline enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java:83]: I'm not convinced that the provider should be tied to one specific analyzer?

**Review by ragadeeshu — CHANGES_REQUESTED** [review]
A few comments, I like that you managed to hide the factory.

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java (hunk 3 above)
Comment (chrisvest) [reply enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java:83]: In the future I want that parameter to be understood as the _default_ analyser, and provide an overload of `createIndex` that also takes an analyser.

File: enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java (hunk 2 above)
Comment (chrisvest) [reply enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/FulltextAnalyzerTest.java]: Yeah, I'll box it up and put it in the fridge.

**Comment by chrisvest** [comment]
@ragadeeshu I believe I addressed your comments, so please take another look.

File: enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java (hunk 3 above)
Comment (ragadeeshu) [reply enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java:83]: I see, that makes sense

**Review by ragadeeshu — APPROVED** [review]

**Pull request merged**